autoload/gen_tags.vim

```diff
@@ -72,7 +72,15 @@ endfunction
 "Prune exit job from job list
 function! s:job_prune(cmd) abort
   for l:item in s:job_list
-    if a:cmd ==# l:item['cmd']
+    let l:cmd = a:cmd
+    let l:item_cmd = l:item['cmd']
+    if type(l:cmd) == type([])
+      let l:cmd = string(l:cmd)
```

Comment:
Use `string()` on a list, still got a list.
Maybe you should use `join()` here?

I checked the source, actually all the source code which calls the `gen_tags#system_async()` is converted to list all ready, except this snippet

```vim
  else
    if has('unix')
      let l:cmd = a:cmd . ' &'
    else
      let l:cmd = 'cmd /c start ' . a:cmd
    endif
```

I think we should change this part to list instead of convert list to string.